components/calendar/src/calendar_arithmetic.rs

```diff
@@ -172,13 +185,13 @@ impl<C: CalendarArithmetic> ArithmeticDate<C> {
     #[inline]
     fn offset_months(&mut self, mut month_offset: i32) {
         while month_offset != 0 {
-            let year_months = C::months_for_every_year(self.year);
+            let year_months = C::months_for_every_year(self.year, self.year_info);
             if self.month as i32 + month_offset > year_months as i32 {
                 self.year += 1;
                 month_offset -= year_months as i32;
             } else if self.month as i32 + month_offset < 1 {
                 self.year -= 1;
-                month_offset += C::months_for_every_year(self.year) as i32;
+                month_offset += C::months_for_every_year(self.year, self.year_info) as i32;
             } else {
                 self.month = (self.month as i32 + month_offset) as u8;
                 month_offset = 0
```

Comment:
Suggestion, down below in `offset_date`: don't call `load_or_compute_info` if the year didn't change